LIS3MDL.class.nut

```diff
@@ -46,12 +46,10 @@ class LIS3MDL {
     function init() {
         // Update the cached scale so that we can convert readings to gauss
         local reg2 = _readRegister(REG_CTL_2);
-        _scale = math.pow(4 , (reg2 >> 5) + 1);
-    }
-
-    function enable(state) {
-        local bits = state ? 0x00 : 0x02;
-        _writeRegister(REG_CTL_3, bits, 0x02);
+        if(reg2 == 0x00) _scale = 4;
```

Comment:
yeah lots of crazy math to avoid conditionals in this lib - I only fixed the ones that were broken.  I can adjust this one too.